### DISCUSSION TITLE
Disabling scientific notation (1e7 -> 10000000)

### DISCUSSION THREAD

**Discussion by adiperdana23** [teamalgoritma/community#77] [Q&A] [answered]
halo, mau bertanya, saya dalam menjalankan Interactive plotting, label di  x , tampilan angkanya tidak bulat (ada unsur "e" nya), contoh : 1000000 ,2000000,dst...
thank you

<img width="1440" alt="Screen Shot 2022-04-30 at 17 36 50" src="https://user-images.githubusercontent.com/100086394/166102146-70d48b6f-8546-499d-ad18-8f7ee5dce8c0.png">

**Comment by onlyphantom** [comment, marked as answer]
Add a chunk at the top:

```r
options(scipen = 999)
```

That command disables printing your results in scientific notation.

If you don't want to change the settings, you can also use format and set `scientific` to FALSE:

```r
format(3248093802, scientific = FALSE)
```

**Reply by adiperdana23** [reply]
thank you, but  i can't make plot, the command is "Error in gsub("\n", br(), a, fixed = TRUE) : 
input string 9 is invalid in this locale".
<img width="1440" alt="Screen Shot 2022-05-01 at 01 30 50" src="https://user-images.githubusercontent.com/100086394/166118125-1af549ee-34b5-48fb-acb1-0be7496597af.png">

**Reply by onlyphantom** [reply]
Can you try to put the options in another chunk (not  in the same chunk, but in a separate chunk above it) just to isolate the error a little more? 

Usually I’ll put the options all the way at the top in its own chunk. Then we can see if that error comes from that chunk or from your ggplot chunk.

Would you also be able to click on Show Traceback and copy + paste the full traceback after moving that options line out?

**Reply by adiperdana23** [reply]
<img width="1440" alt="Screen Shot 2022-05-03 at 12 07 06" src="https://user-images.githubusercontent.com/100086394/166406894-c9f8122f-6b92-4473-89ad-a69cbfba5d60.png">

but I still can't make  interactive plot with ggplotly and the command is 
<img width="792" alt="Screen Shot 2022-05-03 at 12 11 25" src="https://user-images.githubusercontent.com/100086394/166407128-808c2dd4-7ec7-4bae-93e8-6482abdc5bca.png">

**Reply by adiperdana23** [reply]
then i write :

**Sys.setlocale(locale="C")
plot2 <- ggplot(vids_10, aes(x = avg_views, y = reorder(channel_title, avg_views), text = label)) +
  geom_col(aes(fill = avg_views)) +
  labs(title = "Top 10 Channel on Entertainment",
       subtitle = "contoh",
       x = "Average Views",
       y = NULL) +
  scale_fill_gradient(low = "aquamarine", high= "orange") +

  scale_x_continuous(labels = comma) +
  theme_minimal() +
  theme(legend.position = "none")
ggplotly(plot2, tooltip = "text")**

but if I don't input  "Sys.setlocale(locale="C")",  I can't load the flex dashboard.
below my rpubs :

https://rpubs.com/adhiperdanaputra/898028

<img width="1440" alt="Screen Shot 2022-05-04 at 11 12 32" src="https://user-images.githubusercontent.com/100086394/166622261-4acc03a9-cc71-4b3d-ab18-3a8a9fec8bc7.png">

how to ignore **C/C/C/C/C/en_US.UTF-8**

thank you

**Reply by onlyphantom** [reply]
Maybe put your `Sys.setlocale` function call in a separate chunk, probably same chunk as the `options()`. 

If this doesn't solve it, it's better to raise another question in its own thread Pak. When other students ask questions relating to scientific notation, we will point them to this thread (#77) so it's better to have this thread addressing that specifically and another question thread addressing the locale issue.

**Reply by onlyphantom** [reply]
Also: the issue is that there are video names / description in your dataset that uses invalid characters `<br />`, `br()` etc. 

These are not valid characters under the default locale. So when you set locale those goes away. In my opinion, the more appropriate thing to do is to either use a different charset when reading a CSV or / and clean those characters away. 

```r
# if video names include Latin characters for example, we want to change it here
read.csv("youtube.csv", fileEncoding = "latin1")
# otherwise
read.csv("youtube.csv", fileEncoding = "UTF-8")

# then clean / pre-process things that are not part of the character set away,
# for example Emoji, invalid chars like `br()` or `破` (oriental characters) before creating plot
```

**Reply by adiperdana23** [reply]
okay, done.. thank you
